packages/twenty-server/src/engine/core-modules/auth/services/sign-in-up.service.ts

```diff
@@ -280,14 +297,32 @@ export class SignInUpService {
       canImpersonate: boolean;
       canAccessFullAdminPanel: boolean;
     },
+    host?: string,
   ) {
     const userCreated = this.userRepository.create({
       ...newUserWithPicture,
       canImpersonate,
       canAccessFullAdminPanel,
     });
 
-    return await this.userRepository.save(userCreated);
+    const savedUser = await this.userRepository.save(userCreated);
+
+    this.workspaceEventEmitter.emitCustomBatchEvent(
+      USER_SIGNUP_EVENT_NAME,
+      [
+        {
+          userId: savedUser.id,
+          userEmail: newUserWithPicture.email,
+          userFirstName: newUserWithPicture.firstName,
+          userLastName: newUserWithPicture.lastName,
+          locale: newUserWithPicture.locale,
+          host,
```

Comment:
I'm actually challenging the fact that we need the host at this point? The props drilling seems heavy. Why is important? is it to know if it's  sign up on a given workspace? workspaceId (optional) would be nice in this case?

Reply:
We don't have workspaceId at this stage. The goal is to gather more info on self-hosted instances. I ended up using SERVER_URL which isn't as good but I also didn't like the props drilling so that's an OK compromise between business and code